test.ps1

```diff
@@ -2,19 +2,37 @@
 #requires -version 4
 [CmdletBinding(PositionalBinding = $true)]
 param(
-    [Parameter(Mandatory=$true)]
+    [Parameter(Mandatory = $true)]
     [string]$Command,
-    [Parameter(Mandatory=$true)]
+    [Parameter(Mandatory = $true)]
     [string]$RepoPath,
     [switch]$NoBuild = $false,
     [Parameter(ValueFromRemainingArguments = $true)]
-    [string[]]$Arguments
+    [System.Collections.ArrayList]$Arguments
 )
 
 if (!$NoBuild) {
-    & .\build.ps1 /t:PackageKoreBuild
+    & .\build.ps1 /p:SkipTests=$true
 }
 
 $toolsSource = "$PSScriptRoot/artifacts/"
+$latestFile = Join-Path $toolsSource "korebuild/channels/dev/latest.txt"
+$toolsVersion = $null
+foreach ($line in Get-Content $latestFile) {
+    $toolsVersion = $line.Split(":")[1]
+    break
+}
+$packageDir = Join-Path $toolsSource "build\"
+$versionPropsPath = Join-Path $toolsSource "dotnetpackageversion.props"
+$sourcePropsPath = Join-Path $toolsSource "source.props"
+
+$versionPropsValue = "<Project><PropertyGroup><InternalAspNetCoreSdkPackageVersion>$toolsVersion</InternalAspNetCoreSdkPackageVersion></PropertyGroup></Project>"
+$sourcePropsValue = "<Project><PropertyGroup><DotNetRestoreSources>$packageDir</DotNetRestoreSources></PropertyGroup></Project>"
+
+Out-File -FilePath $versionPropsPath -InputObject $versionPropsValue
+Out-File -FilePath $sourcePropsPath -InputObject $sourcePropsValue
+
+$Arguments.Add("/p:DotNetPackageVersionPropsPath=$versionPropsPath")
```

Comment:
Did you try this? 
```
$Arguments += "/p:DotNetPackageVersionPropsPath=$versionPropsPath"
```

Reply:
Ahh, I changed the type because PS complains that Arrays are a fixed size structure when I tried to do `.Add`, but it seems like `+=` has some behind the scenes copy behavior to deal with this.